### ISSUE TITLE
Automation based on lux level works only after opening Home app

### ISSUE THREAD

**Issue by piotrbielaska** [honkmaster/homebridge-mi-flower-care#6]
Hi. First I would like to thank you for great job on creating this plugin. Finally I and, what is most important, my wife has access to Mi Flower readings without need to open Mi Home app.

I did an automation based on lux level readings from the Mi Flower sensor which turns on and off the lights in the living room. It works but the problem is that automation happens only after I open Home app and readings are refreshed. Is it a common thing for current version of Homekit/Homebridge or a bug? 

In a mean time, I'm waiting for the shipment of  Mi Aqara Presence detector, which also have lux level sensor, and I hope I will be able to use it for the same purpose.

**Comment by sorriso93** [comment]
Hello, just curious. How did you get the automation with the lux sensor? I don’t get any action on home app except for moisture switch...

**Comment by piotrbielaska** [comment]
You can use either [Hesperus](https://itunes.apple.com/us/app/hesperus/id969348892?mt=8), [Controller](https://itunes.apple.com/us/app/controller-for-homekit/id1198176727?mt=8&at=1000lHC8) or [Home by Matthias](https://itunes.apple.com/us/app/home-smart-home-automation/id995994352) iOS apps. They allow to use lux level as a trigger to start automation. It is great that you can set not only a value but also a range between values in which automation should be triggered.

**Comment by paqpaqpaq** [comment]
same here
lux and moisture thresholds only work when an app is addressed that contains homekit setup.

**Comment by honkmaster** [comment]
I have no idea why the plugin is behaving this way. Sorry.

**Comment by deadlyknapsack** [comment]
> Hi. First I would like to thank you for great job on creating this plugin. Finally I and, what is most important, my wife has access to Mi Flower readings without need to open Mi Home app.
> 
> I did an automation based on lux level readings from the Mi Flower sensor which turns on and off the lights in the living room. It works but the problem is that automation happens only after I open Home app and readings are refreshed. Is it a common thing for current version of Homekit/Homebridge or a bug?
> 
> In a mean time, I'm waiting for the shipment of Mi Aqara Presence detector, which also have lux level sensor, and I hope I will be able to use it for the same purpose.

I am experiencing the same thing. Did you happen to find a solution? Built an automation, which works fine. However, only when I open the app myself the humidity values get updated in Home app.

**Comment by piotrbielaska** [comment]
I think that there is no solution for this issue. I guess it's a matter of what automations are possible with HomeKit. I decided to start from scratch and use Home Assistant with HomeKit plugin instead. I have a database with history as additional feature. For automation, I use Node-Red. This way I was able to use Mi Flora as a source of lighting level to turn on/off my evening scene. It works perfectly.
